core/protobuf/src/proto/dataserver/protocol.proto

```diff
@@ -5,63 +5,129 @@ package alluxio.proto.dataserver;
 import "dataserver/status.proto";
 
 // The read/write request type. It can either be an Alluxio block operation or a UFS file operation.
-// next available id: 3
+// next available id: 2
 enum RequestType {
   ALLUXIO_BLOCK = 0;
   UFS_FILE = 1;
-  UFS_BLOCK = 2;
 }
 
 // The read request.
-// next available id: 9
+// next available id: 8
 message ReadRequest {
-  optional RequestType type = 1;
-  // The block ID or UFS file ID.
-  optional int64 id = 2;
-  optional int64 offset = 3;
-  optional int64 length = 4;
+  optional int64 block_id = 1;
+  optional int64 offset = 2;
+  optional int64 length = 3;
   // If set, this request is to cancel the reading request for the id.
-  optional bool cancel = 5;
+  optional bool cancel = 4;
+  // Whether the block should be promoted before reading
+  optional bool promote = 7;
 
-  optional int64 lock_id = 6;
-  optional int64 session_id = 7;
-  // If set, do not try to cache the block locally when reading the data from the UFS.
-  optional bool no_cache = 8;
   // If set, the server should send packets in the specified packet size.
-  optional int64 packet_size = 9;
+  optional int64 packet_size = 5;
+
+  // This is only set for UFS block read.
+  optional OpenUfsBlockOptions open_ufs_block_options = 6;
+}
+
+// Options to open a UFS block.
+// next available id: 7
+message OpenUfsBlockOptions {
+  optional string ufsPath = 1;
```

Comment:
should we rename this `ufs_path` from `ufsPath` to keep style consistent?
similar to `maxUfsReadConcurrency` and `mountId`?